lib/swaggerValidator/ajvSchemaValidator.ts

```diff
@@ -192,6 +201,49 @@ export const ajvErrorToSchemaValidateIssue = (
   return result;
 };
 
+const ReValidateIfNeed = (
+  errors: ErrorObject[],
+  ctx: SchemaValidateContext,
+  data: any,
+  validate: ValidateFunction
+): ErrorObject[] => {
+  const result: ErrorObject[] = [];
+  const length = errors.length;
+  const newData = lodash.cloneDeep(data);
+
+  for (let i = 0; i < length; i++) {
+    validate.errors = null;
+    const error = errors[i];
+    const { schema, parentSchema: parentSch, keyword, data: errorData, dataPath } = error;
+    const parentSchema = parentSch as Schema;
+
+    // If the value of query parameter is in string format, we can skip this error
+    if (
+      !ctx.isResponse &&
+      keyword === "type" &&
+      schema === "array" &&
+      typeof errorData === "string" &&
+      (parentSchema as any)?.["in"] === "query"
+    ) {
+      const arrayData = errorData.split(",").map((item) => {
+        if (parseFloat(item)) {
+          return parseFloat(item);
+        }
+        return item;
+      });
+      const position = dataPath.substr(1);
+      lodash.set(newData, position, arrayData);
+      const isValid = validate.call(ctx, newData);
+      if (isValid || !lodash.isEqual(validate.errors![i], error)) {
```

Comment:
When will validate.errors![i] === error?

Reply:
I have updated the logic of this part, which is used to add new errors to result